llvm/lib/Target/RISCV/GISel/RISCVLegalizerInfo.cpp

```diff
@@ -582,10 +582,13 @@ RISCVLegalizerInfo::RISCVLegalizerInfo(const RISCVSubtarget &ST)
       .libcallFor(ST.is64Bit(), {{s32, s128}, {s64, s128}});
 
   // FIXME: We can do custom inline expansion like SelectionDAG.
-  // FIXME: Legal with Zfa.
   getActionDefinitionsBuilder({G_FCEIL, G_FFLOOR, G_FRINT, G_FNEARBYINT,
                                G_INTRINSIC_TRUNC, G_INTRINSIC_ROUND,
-                               G_INTRINSIC_ROUNDEVEN})
+                               G_INTRINSIC_ROUNDEVEN, G_FMAXIMUM, G_FMINIMUM})
```

Comment:
Do we have tests for libcalls for G_FMAXIMUM, G_FMINIMUM without Zfa?

Reply:
No we do not and you also cannot legalize these via libcall either it seems that is a defect :/  

was trying with this snippet, I suppose this change needs more work. I will shift this to a draft

llc mtriple=riscv32 -mattr+f -global-isel %s  
```
define float @fmaximum(float %a, float %b) {
   %c = call float @llvm.maximum.f32(float %a, float %b)
    ret float %c
}
```

Reply:
Just add a separate `getActionDefinitionsBuilder({G_FMAXIMUM, G_FMINIMUM})` that doesn't contain the libcallFor.

Reply:
Yeah this is the best choice for now we can hand the libcall in a different patch set